cpp/src/strings/convert/utilities.cuh

```diff
@@ -67,8 +67,9 @@ __device__ inline size_type integer_to_string(IntegerType value, char* d_buffer)
   bool const is_negative = cuda::std::is_signed<IntegerType>() ? (value < 0) : false;
 
   constexpr IntegerType base = 10;
-  constexpr int MAX_DIGITS   = 20;  // largest 64-bit integer is 20 digits
-  char digits[MAX_DIGITS];          // place-holder for digit chars
+  // largest 64-bit integer is 20 digits; largest 128-bit integer is 39 digits
+  constexpr int MAX_DIGITS = sizeof(IntegerType) > 8 ? 39 : 20;
```

Comment:
```suggestion
  constexpr int MAX_DIGITS = std::numeric_limits<IntegerType>::digits10;
```

Reply:
Unfortunately, this does not give the values we want. 
`cuda::std::numeric_limits<int64_t>::digits10 == 18   (needs to be 20)`
`cuda::std::numeric_limits<__int128_t>::digits10 == 38   (needs to be 39)`

Reply:
Oh yeah, you always need to do +1 to get the total number of digits because `digits10` is the number of digits it can hold _without loss_. So the last digit isn't counted since it can't go from 0...9 without hitting the max value.

Why is 20 the max digits for int64? Do you mean uint64? In which case the +1 should work for all types. 

https://godbolt.org/z/Yc78EqYnG

```suggestion
  constexpr int MAX_DIGITS = cuda::std::numeric_limits<IntegerType>::digits10 + 1;
```

Reply:
I am not sure if this works for @davidwendt's new code, but in `count_digits` there is a similar piece of code that I wrote:
```cpp
auto constexpr max_digits = std::is_same_v<IntegerType, __int128_t> ? 39 : 20;
```
and then that is replaced with:
```cpp
auto constexpr max_digits = cuda::std::numeric_limits<IntegerType>::digits10 + 1;
```
it works. So I would recommend changing that code as well.

Reply:
The `digits10 + 1` is working for all `int` types. I changed the code in both places.

Reply:
:fire: